lua/nvim-tree/explorer/sorters.lua

```diff
@@ -174,12 +180,14 @@ function C.suffix(a, b)
   end
 
   -- directories go first
-  if a.nodes and not b.nodes then
-    return true
-  elseif not a.nodes and b.nodes then
-    return false
-  elseif a.nodes and b.nodes then
-    return C.name(a, b)
+  if M.config.sort_folders_first then
+    if a.nodes and not b.nodes then
+      return true
+    elseif not a.nodes and b.nodes then
+      return false
+    elseif a.nodes and b.nodes then
+      return C.name(a, b)
```

Comment:
@moniquelive should we sort folders by suffix? Does that make any sense?

Reply:
it's not very usual, but sometimes you may have folders with suffixes to disambiguate their names such as `.tmp`, `.old`, `.bkp`, etc. But I believe people will want to turn this directory-grouping-by-suffix on or off. Should this be another setting? This is quickly getting fractal... 👀

Reply:
We have too many options... I'm happy for this to be opinionated.

Other sort methods apply the same rules to folders and files, this should too.

Reply:
Tried it, it looked odd, abandoned.